### PR TITLE
Wildcard

### PR DESCRIPTION
**Pull request by rfechney**
Here's the wildcard branch with reconnect functionality.  Run it up and see how it goes.

### CODE REVIEW

File: README.md
Comment (joebywan) [inline README.md:1]: So build's tested on:

* ubuntu 20.04
* 22.04
* 24.04

I'm testing on Linux Mint 21.3

Just figure we should be careful with what we're stating.

**Review by joebywan — COMMENTED** [review]
Gave it a test, the virtual device stays, but does not reconnect. Journalctl doesn't show any output as to why.

Test procedures:
install update
connect tpr
start machine
test functionality of all axes with antimicrox. 

Result: Both phys and virt devices respond with virt showing the button.

Disconnect tpr
test functionality of all axes with antimicrox.

Result: phys device not present, virt present.

connect tpr
test functionality of all axes with antimicrox.

Result: both phys and virt present, phys responsive to all axes, virt unresponsive.

Also tried:
connecting after machine start - virt not responsive.
Manual stop/start of service does regain functionality as expected.

**Comment by rfechney** [comment]
Ok.  I guess it is not detecting the re-connection, or perhaps it never detects the drop out in the first place.

I'll see if I can figure it out.

**Comment by rfechney** [comment]
I pushed a change to reset the input device connection if it gets a bad return code.  Hopefully that will detect a disconnect.

The virtual device should persist if the system is running.  If we want to change that we can though.

File: src/protontpr.cpp
Comment (joebywan) [inline src/protontpr.cpp:1]: The reconnect isn't working, but also virtual device connectivity has been interrupted.

The virtual device shows up, but it doesn't respond to input in antimicrox

journalctl -u protontpr.service -f shows "In use Thrustmaster T-Pendular-Rudder device /dev/input/by-id/usb-Thrustmaster_T-Pendular-Rudder-event-joystick returns -11 on read, resetting."

**Pull request merged**